src/CoreWCF.Primitives/src/CoreWCF/Description/TypeLoader.cs

```diff
@@ -1117,9 +1119,74 @@ private void VerifyConsistency(OperationConsistencyVerifier verifier)
                     requestDescription.Body.WrapperName = requestDescription.Body.WrapperNamespace = null;
                 }
             }
+
+            var serviceImplementationMethodInfo = FindServiceImplementationMethodInfo(operationDescription, contractDescription);
+            if (serviceImplementationMethodInfo == null)
+            {
+                // we do care only about Authorize attributes that decorate non callback contract
+                Fx.AssertAndThrow(direction == MessageDirection.Output,
+                    $"Unable to find service implementation in contract {contractDescription.Name}");
+            }
+            else
+            {
+                methodAttributes = serviceImplementationMethodInfo.GetCustomAttributes(false);
+                foreach (var methodAttribute in methodAttributes.OfType<IAuthorizeData>())
+                {
+                    operationDescription.AuthorizeData.Add(methodAttribute);
+                }
+            }
+
             return operationDescription;
         }
 
+        private static MethodInfo FindServiceImplementationMethodInfo(OperationDescription operationDescription, ContractDescription contractDescription)
+        {
```

Comment:
There's a lot easier way to do this using an [interface map](https://learn.microsoft.com/en-us/dotnet/api/system.type.getinterfacemap?view=net-6.0). I think code like this should work:
```c#
Type serviceType = typeof(TService);
var ifaceMap = serviceType.GetInterfaceMap(operationDescription.OperationMethod.DeclaringType);
var index = Array.IndexOf(ifaceMap.InterfaceMethod, operationDescription.OperationMethod);
if (index < 0) return null;
return ifaceMap.TargetMethods[index];
```

Reply:
It works but break 2 tests:
- one test named `InterfaceOnlyServiceTest` that register service in DI using interface and pull implementation using `ServiceProvider`.
- another one named `DuplexEcho`

Reply:
This seems to work
```c#
            Type serviceType = typeof(TService);
            if (serviceType.IsInterface || operationDescription.OperationMethod.DeclaringType!.IsInterface)
            {
                return operationDescription.OperationMethod;
            }

            InterfaceMapping interfaceMapping = serviceType.GetInterfaceMap(operationDescription.OperationMethod.DeclaringType);
            int index = Array.IndexOf(interfaceMapping.InterfaceMethods, operationDescription.OperationMethod);
            return index < 0
                ? null
                : interfaceMapping.TargetMethods[index];
```

Reply:
done here https://github.com/CoreWCF/CoreWCF/pull/850/commits/719999fb8b5771d6060ce4d2e59a307406415718

Reply:
Thanks i did not know `InterfaceMapping`!

Reply:
This means this feature won't work when CoreWCF doesn't know the type ahead of time.

Reply:
When the "Fully register CoreWCF service with interface" has been added it assumed the concrete service type will be in DI. What do you think about trying to pull an instance from DI using a ServiceScope, getting the concrete type and pull the attributes applied to it ?

Reply:
@mconnew I brought support of Authorize at ServiceContract level and dropped the analyzer that was emitting build errors